tests/api-watch-diff/test.ml

```diff
@@ -2,3 +2,198 @@ let%expect_test "test_diff_interface" =
+        type_separability = [];
+        type_is_newtype = false;
+        type_expansion_scope = 0;
+        type_loc = Location.none;
+        type_attributes = [];
+        type_immediate = Unknown;
+        type_unboxed_default = false;
+        type_uid = Uid.internal_not_actually_unique;
+      },
+      Trec_not,
+      Exported )
+
+(* Signature for type t:
+     > type unused_type = string *)
+
+let unused_type_sig =
+  Sig_type
+    ( Ident.create_persistent "unused_type",
+      {
+        type_params = [];
+        type_arity = 0;
+        type_kind = Type_abstract;
+        type_private = Public;
+        type_manifest =
+          Some
+            (create_expr
+               (Tconstr (Predef.path_string, [], ref Mnil))
+               ~level:0 ~scope:0 ~id:1);
+        type_variance = [];
+        type_separability = [];
+        type_is_newtype = false;
+        type_expansion_scope = 0;
+        type_loc = Location.none;
+        type_attributes = [];
+        type_immediate = Unknown;
+        type_unboxed_default = false;
+        type_uid = Uid.internal_not_actually_unique;
+      },
+      Trec_not,
+      Exported )
+
+(* Signature for value f:
+    > val f : t -> string *)
+
+let val_f_sig =
+  Sig_value
+    ( Ident.create_persistent "f",
+      {
+        val_type =
+          create_expr
+            (Tarrow
+               ( Nolabel,
+                 create_expr
+                   (Tconstr
+                      (Path.Pident (Ident.create_persistent "t"), [], ref Mnil))
+                   ~level:0 ~scope:0 ~id:2,
+                 create_expr
+                   (Tconstr (Predef.path_string, [], ref Mnil))
+                   ~level:0 ~scope:0 ~id:3,
+                 commu_ok ))
+            ~level:0 ~scope:0 ~id:4;
+        val_kind = Val_reg;
+        val_loc = Location.none;
+        val_attributes = [];
+        val_uid = Uid.internal_not_actually_unique;
+      },
+      Exported )
+
+(* Signature for value g:
+    > val g : t -> t *)
+
+let val_g_sig =
+  Sig_value
+    ( Ident.create_persistent "g",
+      {
+        val_type =
+          create_expr
+            (Tarrow
+               ( Nolabel,
+                 create_expr
+                   (Tconstr
+                      (Path.Pident (Ident.create_persistent "t"), [], ref Mnil))
+                   ~level:0 ~scope:0 ~id:5,
+                 create_expr
+                   (Tconstr
+                      (Path.Pident (Ident.create_persistent "t"), [], ref Mnil))
+                   ~level:0 ~scope:0 ~id:6,
+                 commu_ok ))
+            ~level:0 ~scope:0 ~id:7;
+        val_kind = Val_reg;
+        val_loc = Location.none;
+        val_attributes = [];
+        val_uid = Uid.internal_not_actually_unique;
+      },
+      Exported )
+
+(* Signature for file ref.mli:
+      > type t = int
+      > type unused_type = string
+      > val f : t -> string *)
+
+let ref_signature = [ t_sig; unused_type_sig; val_f_sig ]
```

Comment:
This is a really nice approach, and it saves having to write a bunch of duplicate code. It makes your tests very easy to read.

Reply:
Thanks @shonfeder :)